server/persistence/readclient/readclient_list_test.go

```diff
@@ -29,6 +34,129 @@ var _ = Describe("List", func() {
 		rc = readclient.New(frc, mp)
 	})
 
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	DescribeTable("Filter by label", func(unfilteredInternalWorkspaces []workspacesv1alpha1.InternalWorkspace, expectedObjectMetas []metav1.ObjectMeta) {
+		// given
+		user := "user"
+		// internalLabel := workspacesv1alpha1.LabelInternalDomain + "whatever"
+
+		// internal client expected to be called once.
+		// It returns no error so we can test the filtering by label.
+		frc.EXPECT().
+			ListAsUser(gomock.Any(), gomock.Any(), gomock.Any()).
+			DoAndReturn(func(_ context.Context, _ string, iww *workspacesv1alpha1.InternalWorkspaceList) error {
+				iww.Items = unfilteredInternalWorkspaces
+				return nil
+			}).
+			Times(1)
+
+			// mapper expects to be called once.
+			// It returns the list of mapped workspaces (just objectmeta)
+		mp.EXPECT().
+			InternalWorkspaceListToWorkspaceList(gomock.Any()).
+			Times(1).
+			DoAndReturn(func(iww *workspacesv1alpha1.InternalWorkspaceList) (*restworkspacesv1alpha1.WorkspaceList, error) {
+				ww := restworkspacesv1alpha1.WorkspaceList{Items: []restworkspacesv1alpha1.Workspace{}}
+				for _, w := range iww.Items {
+					ww.Items = append(ww.Items, restworkspacesv1alpha1.Workspace{ObjectMeta: w.ObjectMeta})
+				}
+				return &ww, nil
+			})
+
+		// when
+		actualWorkspaces := restworkspacesv1alpha1.WorkspaceList{}
+		err := rc.ListUserWorkspaces(ctx, user, &actualWorkspaces, client.MatchingLabels{"hit": "hit"})
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+		Expect(actualWorkspaces.Items).To(HaveLen(len(expectedObjectMetas)))
+		oww := make([]metav1.ObjectMeta, len(actualWorkspaces.Items))
+		for i, w := range actualWorkspaces.Items {
+			oww[i] = w.ObjectMeta
+		}
+		Expect(oww).To(BeEquivalentTo(expectedObjectMetas))
```

Comment:
@sadlerap do you have any suggestions here?

Reply:
I have a way of doing this without copying the `ObjectMeta` fields out, but it's longer than this.  This is more than fine for now.